### PR TITLE
[IMP] core: odoo.domain

### PR DESCRIPTION
**Pull request by kmagusiak**
odoo/odoo#170009

### CODE REVIEW

**Review by rco-odoo — APPROVED** [review]

File: content/developer/reference/backend/security.rst [hunk 1]
```diff
@@ -293,6 +310,30 @@ online documentation of pyscopg2 to learn of to use it properly:
...
 - `Psycopg documentation <https://www.psycopg.org/docs/sql.html>`_
 
+Building the domains
+--------------------
+
+Domains are represented as lists and are serializable.
+You would be tented to manipulate these lists directly, however it can
+introduce subtle issues where the user could inject a domain if the input is
+not normalized.
+We introduce :class:`~odoo.fields.Domain` to handle safely the manipulation
```
Comment (AntoineVDV) [inline content/developer/reference/backend/security.rst:320]:
```suggestion
Use :class:`~odoo.fields.Domain` to handle safely the manipulation
```

File: content/developer/reference/backend/orm.rst [hunk 2]
```diff
@@ -968,32 +971,56 @@ A domain is a list of criteria, each criterion being a triple (either a
...
+        ``field_expr`` (:class:`~odoo.fields.Many2one`,
         :class:`~odoo.fields.One2many`, or :class:`~odoo.fields.Many2many`)
         satisfies the provided domain ``value``.
 
 * ``value``
     variable type, must be comparable (through ``operator``) to the named
     field.
 
-Domain criteria can be combined using logical operators in *prefix* form:
+The :class:`~odoo.fields.Domain` can be used as a builder for domains.
```
Comment (AntoineVDV) [inline content/developer/reference/backend/orm.rst:988]: Either
```suggestion
:class:`~odoo.fields.Domain` can be used as a builder for domains.
```
but then it'd be better to auto-document it, or
```suggestion
The :class:`~odoo.fields.Domain` class can be used as a builder for domains.
```
or 
```suggestion
The :class:`~odoo.fields.Domain` constructor can be used as a builder for domains.
```
or
```suggestion
The :class:`~odoo.fields.Domain` API can be used as a builder for domains.
```
... you get it.

**Review by AntoineVDV — COMMENTED** [review]
All of this is correct both factually and RST-wise, but I find it a bit confusing for the reader because `Domain` is never properly introduced as a builder|wrapper|API (not even sure about what it is) for search domains.

It starts as if importing `odoo.fields.Domain` and then using it to declare domains is the way to go, but immediately after, the reference doc is all about creating domains in list representation.

**Comment by kmagusiak** [comment]
I applied the suggestions.

> All of this is correct both factually and RST-wise, but I find it a bit confusing for the reader because `Domain` is never properly introduced as a builder|wrapper|API (not even sure about what it is) for search domains.

In the "search domains" section, there is
`:class:`~odoo.fields.Domain` can be used as a builder for domains.`.

> It starts as if importing `odoo.fields.Domain` and then using it to declare domains is the way to go, but immediately after, the reference doc is all about creating domains in list representation.

I'm not sure what you mean. Just after we introduce it, there are some examples using the class. They are followed by some examples that use the serialized (list) form.
Both are supported.

**Review by rco-odoo — APPROVED** [review]
Although it is not perfect, the documentation of domains looks good enough to me.  @AntoineVDV can we please merge it, and not block the linked pull requests?  We can improve the documentation later.

**Comment by AntoineVDV** [comment]
> Although it is not perfect, the documentation of domains looks good enough to me. @AntoineVDV can we please merge it, and not block the linked pull requests? We can improve the documentation later.

Woops, I didn't mean to block it, or I would have stated it clearly; I just forgot the delegate :zany_face: 

@robodoo r+ delegate+

Pro tip: If a doc PR is blocking attached PRs, it's always possible to close the doc PR during the merge and reopen it after.

**Pull request closed without merging**